tests/integration/iop/test_dag_process_file.py

```diff
@@ -0,0 +1,149 @@
+
+    return enriched_file
+
+
+def test_dag_loaded(dag: DAG):
+    assert dag is not None
+    assert len(dag.tasks) == 4
+
+
+def test_dag_validate_file_processing(article):
+    iop_validate_record(article)
+
+
+publisher = "IOP"
+
+generic_pseudo_parser_output = {
+    "abstract": "this is abstracts",
+    "copyright_holder": "copyright_holder",
+    "copyright_year": "2020",
+    "copyright_statement": "copyright_statement",
+    "copyright_material": "copyright_material",
+    "date_published": "2022-05-20",
+    "title": "title",
+    "subtitle": "subtitle",
+}
+
+
+expected_output = {
+    "abstracts": [{"value": "this is abstracts", "source": publisher}],
+    "acquisition_source": {
+        "source": publisher,
+        "method": publisher,
+        "date": "2022-05-20T00:00:00",
+    },
+    "copyright": [
+        {
+            "holder": "copyright_holder",
+            "year": "2020",
+            "statement": "copyright_statement",
+            "material": "copyright_material",
+        }
+    ],
+    "imprints": [{"date": "2022-05-20", "publisher": publisher}],
+    "record_creation_date": "2022-05-20T00:00:00",
+    "titles": [{"title": "title", "subtitle": "subtitle", "source": publisher}],
+}
+
+empty_generic_pseudo_parser_output = {
+    "abstract": "",
+    "copyright_holder": "",
+    "copyright_year": "",
+    "copyright_statement": "",
+    "copyright_material": "",
+    "date_published": "",
+    "title": "",
+    "subtitle": "",
+}
+
+
+expected_output_from_empty_input = {
+    "abstracts": [{"value": "", "source": publisher}],
+    "acquisition_source": {
+        "source": publisher,
+        "method": publisher,
+        "date": "2022-05-20T00:00:00",
+    },
+    "copyright": [{"holder": "", "year": "", "statement": "", "material": ""}],
+    "imprints": [{"date": "", "publisher": publisher}],
+    "record_creation_date": "2022-05-20T00:00:00",
+    "titles": [{"title": "", "subtitle": "", "source": publisher}],
+}
+
+
+@pytest.mark.parametrize(
+    "test_input, expected, publisher",
+    [
+        pytest.param(generic_pseudo_parser_output, expected_output, publisher),
+        pytest.param(
+            empty_generic_pseudo_parser_output,
+            expected_output_from_empty_input,
+            publisher,
+        ),
+    ],
+)
+@freeze_time("2022-05-20")
+def test_dag_enhance_file(test_input, expected, publisher):
+    assert expected == iop_enhance_file(test_input)
+
+
+def test_dag_enrich_file(assertListEqual):
+    input_article = {
+        "arxiv_eprints": [{"value": "2112.01211"}],
+        "curated": "Test Value",
+        "citeable": "Test Value",
+        "files": "Test Value",
+    }
+    assertListEqual(
+        {
+            "arxiv_eprints": [
+                {"value": "2112.01211", "categories": list(set(["hep-th", "hep-ph"]))}
+            ],
+            "$schema": "http://repo.qa.scoap3.org/schemas/hep.json",
+        },
+        iop_enrich_file(input_article),
+    )
```

Comment:
just a question, cause don't know if it's possible: can't you test the whole dag processing pipeline `iop_process_file`?

Reply:
I actually cab but just to check the state of the DAG: is it "success", "running", "queued", "failed".
Something similar to this:
https://github.com/cern-sis/workflows/blob/5f3a04512c90045cc1898f3bbfe3830160e6ba02/tests/units/aps/test_trigger_files_processing.py#L59-L76

Since we are not saving the ouput of this DAG is hard to test it differently, that's why I am testing every step separately